plugins/carbon_black_cloud/icon_carbon_black_cloud/tasks/monitor_alerts/task.py

```diff
@@ -256,7 +258,12 @@ def get_observations(
             # this is because observations can arrive before the start time and also occur so frequently
             # that they can be returned in a page size but not move the time forward
             if state.get(LAST_OBSERVATION_TIME) <= start_observation_time:
-                state[OBSERVATION_JOB_OFFSET] = state.get(OBSERVATION_JOB_OFFSET, 0) + page_size
+                # if there is no observations found and the job time has been exceeded,
+                # then we want to leave the offset at the same value to ensure there is no gaps
+                if not observations and job_time_exceeded:
```

Comment:
Code lgtm - just wondering if we should log when we hit this? If we're constantly hitting this condition it would just endlessly loop. We could then set up an alert for it?

Reply:
A logger has been added